### PR TITLE
[iOS] Add possibility to disable buttons in action sheet ios

### PR DESCRIPTION
**Pull request by lukewalczak**
## Summary

_**Fixed**_ version of [the previous PR](https://github.com/facebook/react-native/pull/28792) after reverting [changes](https://github.com/facebook/react-native/commit/c8d678abcf93fd3f6daf4bebfdf25937995c1fdf#commitcomment-39299254)
 
----

I've noticed that currently there is no option to disable button within the `ActionSheetIOS`. It can be really useful and decided to extend the API to support that functionality.

I added a new option called `disabledButtonsIndices` to `ActionSheetIOS` which is an array of button indices which should be disabled.

`ActionSheetIOS` documentation - PR https://github.com/facebook/react-native-website/pull/1898

## Changelog

[iOS] [Added] - Add disableButtonsIndices option to ActionSheetIOS component

## Test Plan

1. Run the `RNTester`
2. Choose `ActionSheetIOS`
3. Check the fourth example `Show Action Sheet with disabled buttons`
4. `Option 1` and `Option 2` should be disabled

screenshot | gif
--- | ---
Screenshot 2020-04-30 at 15 16 22 |

### CODE REVIEW

File: RNTester/js/examples/ActionSheetIOS/ActionSheetIOSExample.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const ScreenshotManager = NativeModules.ScreenshotManager;
 const BUTTONS = ['Option 0', 'Option 1', 'Option 2', 'Delete', 'Cancel'];
 const DESTRUCTIVE_INDEX = 3;
 const CANCEL_INDEX = 4;
+const DISABLED_BUTTON_INDICES = [1.0, 2.0];
```
Comment (Simek) [inline RNTester/js/examples/ActionSheetIOS/ActionSheetIOSExample.js:28]: Is there a reason why you have used the float numbers (not ints) in the example?

Comment (lukewalczak) [reply RNTester/js/examples/ActionSheetIOS/ActionSheetIOSExample.js:28]: Nope, will refactor it to ints, thanks!

**Comment by grabbou** [comment]
CC: @shergin is there any update on this?

**Comment by vonovak** [comment]
@shergin would you please take a look at this? Thanks!

to make the review easier, the line that fixes the [crash](https://github.com/facebook/react-native/commit/c8d678abcf93fd3f6daf4bebfdf25937995c1fdf#commitcomment-39299254) is `if (options.disabledButtonIndices())` in https://github.com/facebook/react-native/blob/7d82791c6877dc1e3bb3da8bf5ceee1bd720a5b8/React/CoreModules/RCTActionSheetManager.mm#L80

**Comment by vonovak** [comment]
@lukewalczak there is a conflict, would you mind fixing it? Then we can try ping someone from fb again, thank you!

**Comment by lukewalczak** [comment]
Thanks @vonovak for friendly ping, I've resolved a conflict. @shergin is there something I can do to proceed with that PR?

**Pull request closed without merging**